Add runtime changes to 2201.2.0 release note (#2081)
ballerina-platform/ballerina-release: release-notes/swan-lake-release-notes/swan-lake-2202.2.0-release-note.md

### Runtime updates

#### New features

##### Strand dump

> Allows to get the status of strands and strand groups during the execution of a Ballerina program.

Reviewer: ```suggestion
Allows getting the status of strands and strand groups during the execution of a Ballerina program.
```

This can be used to troubleshoot runtime errors. Ballerina runtime will emit the strand dump to the standard output stream in the text format, if it receives a `SIGTRAP` signal. (`SIGTRAP` is not available on Windows.)

> e.g., If the PID of the running Ballerina program is `$PID`, you can get the strand dump by executing either `kill -SIGTRAP $PID` or `kill -5 $PID` command.

Reviewer: ```suggestion
E.g., if the PID of the running Ballerina program is `$PID`, you can get the strand dump by executing either `kill -SIGTRAP $PID` or `kill -5 $PID` command.
```

> ##### StopHandler

Reviewer: ```suggestion
##### `StopHandler`
```
Shall we specify what this is (i.e., `StopHandler` function or module likewise)?

> Allows to register a function that will be called during graceful shutdown.

Reviewer: ```suggestion
Allows registering a function that will be called during a graceful shutdown.
```

> A call to `onGracefulStop` will result in one call to the handler function that was passed as an argument; the handler functions will be called after calling `gracefulStop` on all registered listeners, in reverse order of the corresponding calls to `onGracefulStop`.

Reviewer: ```suggestion
A call to `onGracefulStop` will result in one call to the handler function that was passed as an argument; the handler functions will be called after calling `gracefulStop` on all registered listeners, in the reverse order of the corresponding calls to `onGracefulStop`.
```

> e.g., A function `foo` can be called during the graceful shutdown by registering it as follows,

Reviewer: ```suggestion
E.g., a `foo` function can be called during the graceful shutdown by registering it as follows.
```

`runtime:onGracefulStop(foo);`

#### Improvements

> #### New Runtime Java APIs

Reviewer: ```suggestion
#### New runtime Java APIs
```
This section seems to be empty. Please add the content or remove it if it is not required.

> ##### Type reference type support at runtime

Reviewer: ```suggestion
##### Type reference type support at runtime
```
This section seems to be empty. Please add the content or remove it if it is not required. Also, the repetition of the word `type` makes it a bit confusing. Shall we try rewording the title?

Reviewer 2: Under the `Type reference type support at runtime` section, we have two sub-sections. 
- `Modified existing runtime APIs`
- `New runtime Java API`
So I think we can keep the title and change it to `Type reference support at runtime`

Author: The section `Type reference type support at runtime` have subsections.

###### Modified existing runtime APIs

> When a type is defined referring to another type, it will be now passed to runtime as a `BTypeReferenceType` instance.

Reviewer: ```suggestion
When a type is defined referring to another type, it will now be passed to the runtime as a `BTypeReferenceType` instance.
```

> For example, the following code contains type reference types `Integer` and `Student`.

Reviewer: ```suggestion
For example, the following code contains the `Integer` and `Student` type reference types.
```

```ballerina
type Integer int;

type Person record {|
    string name;
    Integer age;
|};

type Student Person;

```

> The following runtime Java APIs are now supported to return `BTypeReferenceType` instances.

Reviewer: ```suggestion
The following runtime Java APIs are now supported to return the `BTypeReferenceType` instances.
```

```java

> // from ArrayType

Reviewer: ```suggestion
// from the `ArrayType`.
```

Type getElementType();

> // from FunctionType

Reviewer: ```suggestion
// from the `FunctionType`
```

Parameter[] getParameters();
Type[] getParameterTypes();

> // from Field

Reviewer: ```suggestion
// from the `Field`
```

Type getFieldType();

> // from BTypedesc

Reviewer: ```suggestion
// from the `BTypedesc`
```

Type getDescribingType();
```

###### New runtime Java API

The follwing new runtime APIs are added to provide the referred type of a type reference type.  

> `TypeUtils.getReferredType()`  

Reviewer: ```suggestion
- `TypeUtils.getReferredType()`  
```

> `getReferredType()` in `ReferenceType`  

Reviewer: ```suggestion
- `getReferredType()` in `ReferenceType`  
```

```ballerina
type Integer int;

type Quantity Integer;
```  

> In the above sample for the type `Quantity`,  

Reviewer: ```suggestion
In the above example of the `Quantity` type,  
```

> The `TypeUtils.getReferredType()` call will return the `int` type instance.

Reviewer: ```suggestion
the `TypeUtils.getReferredType()` call will return the `int` type instance.
```

> `getReferredType()` call on the `ReferenceType` will return another `BTypeReferenceType` instance with name `Integer`.

Reviewer: ```suggestion
The `getReferredType()` call on the `ReferenceType` will return another `BTypeReferenceType` instance with the `Integer` name.
```
